specifyweb/frontend/js_src/lib/components/QueryBuilder/FieldFilter.tsx

```diff
@@ -61,9 +68,234 @@ export const filtersWithDefaultValue = new Set<QueryFieldFilter>([
   'in',
 ]);
 
+const relativeDateItems = {
+  direction: [
+    { value: '+', title: queryText.future() },
+    {
+      value: '-',
+      title: queryText.past(),
+    },
+  ],
+  type: [
+    {
+      value: 'day',
+      title: queryText.day(),
+    },
+    { value: 'week', title: queryText.week() },
+    {
+      value: 'month',
+      title: queryText.month(),
+    },
+    {
+      value: 'year',
+      title: queryText.year(),
+    },
+  ],
+};
+
+function PickListSimple({
+  pickListItems,
+  value,
+  onBlur: handleBlur,
+  onChange: handleChange,
+}: {
+  readonly pickListItems: RA<PickListItemSimple>;
+  readonly value: string;
+  readonly onBlur: ({ target }: React.ChangeEvent<HTMLSelectElement>) => void;
+  readonly onChange: (value: string) => void;
+}): JSX.Element {
+  return (
+    <Select value={value} onBlur={handleBlur} onValueChange={handleChange}>
+      {pickListItems.map(({ title, value }) => (
+        <option key={value} value={value}>
+          {title}
+        </option>
+      ))}
+    </Select>
+  );
+}
+
+function extractValuesSimple<TYPE>(
+  target: HTMLInputElement | HTMLSelectElement
+): TYPE {
+  return target.value as unknown as TYPE;
+}
+
+function DateSplit({
+  onChange: handleChange,
+  parsed,
+  onLiveChange: handleLiveChange,
+}: {
+  readonly parsed: {
+    readonly direction: string;
+    readonly type: string;
+    readonly size: number;
+  };
+  readonly onChange: ((newValue: string) => void) | undefined;
+  readonly onLiveChange: (() => void) | undefined;
+}): JSX.Element {
+  const [values, setValues] = useTriggerState<{
+    readonly direction: string;
+    readonly type: string;
+    readonly size: number;
+  }>(parsed);
+  const direction = values.direction;
+  const size = values.size;
+  const type = values.type;
+
+  return (
+    <div className="flex flex-row gap-1">
+      <PickListSimple
```

Comment:
This is an over-abstraction. Directly calling <Select> and <option> two times (rather than in a loop!) simplifies things